docs/api.md

```diff
@@ -53,13 +53,79 @@ In our experience, any system that is successful needs to grow and change as new
 
 What constitutes a compatible change and how to change the API are detailed by the [API change document](devel/api_changes.md).
 
+
+## API groups
+
+To make it easier to extend the Kubernetes API, we are in the process of implementing *API groups*.
+The API group is specified in a REST path and in the `apiVersion` field of a serialized object.
+
+Currently there are two API groups in use:
+
+1. the "core" group, which is at REST path `/api/v1` and is not specified as part of the `apiVersion` field, e.g.
+   `apiVersion: v1`.
+1. the "experimental" group, which is at REST path `/apis/experimental/$VERSION`, and which uses
+  `apiVersion: experimental/$VERSION`, where currently `apiVersion: experimental/v1alpha1`.
+  The name `experimental` implies that API groups themselves are currently under development.  The actual
+  stability of the version of objects within that group is determined by its [API
+  version](#api-versioning)
+
+In the future we expect that there will be more API groups, all at REST path `/apis/$API_GROUP` and
+using `apiVersion: $API_GROUP/$VERSION`.  We expect that there will be a way for third parties to
+create their own API groups, and to avoid naming collisions.
+
+Not all kubernetes clusters will support all API groups.  However, we intend that if an API server
+supports a group, it supports all the Kinds in the group, and all of them at the same version(s).
+
 ## API versioning
 
-To make it easier to eliminate fields or restructure resource representations, Kubernetes supports multiple API versions, each at a different API path prefix, such as `/api/v1beta3`. These are simply different interfaces to read and/or modify the same underlying resources. In general, all API resources are accessible via all API versions, though there may be some cases in the future where that is not true.
+To make it easier to eliminate fields or restructure resource representations, Kubernetes supports multiple API versions, each at a different API path prefix, such as `/api/v1beta3`.
```

Comment:
Moved up.  I didn't like the sentence about *all API resources are accessible via all API* since that is no longer true.  I didn't mean to delete *These are simply different interfaces...underlying resources* and have reinstated that sentence.